### DISCUSSION TITLE
Document Writer v4.5.0 question

### DISCUSSION THREAD

**Discussion by GWhiteGE** [nextgenhealthcare/connect#6171] [Q&A]
I am relatively new to Mirth and was recently asked to attempt extracting data from HL7 into a PDF and sending the file to a shared network drive. I have figured out the PDF creation piece of this using the Document Writer destination connector but can only seem to get it to output to a local folder. Thanks in advance for any advice.

**Comment by pacmano1** [comment]
Either use the file wrtier of type "SMB" or mount the file system local to the mirth server.

I assuming you mean SMB / CIFS (i.e. Windows), not NFS or some other shared file system.

**Reply by pacmano1** [reply]
![image](https://github.com/nextgenhealthcare/connect/assets/44065187/1dc932ba-f41f-4e12-96db-201d2f20d5f7)

**Comment by Hueli** [comment]
A "Document Writer" only offers the option of writing to a directory (or share). The service account under which the service runs must have access to the share.

If Mirth is installed on a Windows system: After installation (and unfortunately also after every Mirth update) the service runs with the “Local System” account. After restarting the service with a suitable service account, access in the "Document Writer" is possible via UNC path.

Alternatively, you could assign a drive letter in the channel's deploy script using a "net use" command with the user data (password in plain text!) and then insert this letter with path in the "Directory" field of the document writer ("net use / delete" in undeploy script). However, this is not very elegant and insecure.

It is better to run the Mirth service itself with an account that has access to the share and then address this as a UNC path in the "Document Writer".

**Reply by pacmano1** [reply]
We are talking about two things here of course. One is the creation of the document itself. Two is writing the file to a file system.  in general, I would typically create an attachment (if I'm using the in-line document writer) to create a PDF then on another step write that attachment using the file writer.

**Reply by GWhiteGE** [reply]
So with the SMB method in File Writer, can the same HTML Template I created in Document Writer be used as the Template in File Writer, and would File Type then be set to Binary?

**Reply by pacmano1** [reply]
On the first destination, you would create the PDF as an attachment. On the second destination, you would write that attachment using the file writer.

**Reply by Hueli** [reply]
I'm sorry, but the question was specifically about the "Document Writer". Of course, you can also output the resulting document via a second channel and then via a “File Writer”. But that wasn't asked. And everyone does it the way they think is best. Many roads lead to Rome, some short and some long.

**Reply by pacmano1** [reply]
@Hueli - I retracted my first answer about what is best already - that was my bad.   Yes, your point is correct if you want to do it one step and the mirth server runs on Windows.

A few notes:

1. UNC works on Windows only.  The OP did not say what host OS mirth was on.  Nor did they say if it was SMB or NFS.
2. Linux has all kinds of work arounds to workaround users and mounted file structures such that the mirth process owner can be irrelevant depending on how the file structure is mounted.
3. If containerizing and the network share is SMB, I think most folks would probably use the File Write of type SMB.
